assets/javascripts/modules/checkout/toggleFieldsets.js

```diff
@@ -0,0 +1,116 @@
+define([
+    'bean',
+    'utils/ajax',
+    'modules/forms/toggleError',
+    'modules/checkout/formElements',
+    'modules/checkout/validations'
+], function (
+    bean,
+    ajax,
+    toggleError,
+    formEls,
+    validations
+) {
+    'use strict';
+
+    var FIELDSET_COLLAPSED = 'fieldset--collapsed';
+    var FIELDSET_COMPLETE = 'data-fieldset-complete';
+    var IS_HIDDEN = 'is-hidden';
+
+    function swap(from, to) {
+        from.addClass(IS_HIDDEN);
+        to.removeClass(IS_HIDDEN);
+    }
+
+    function collapseFieldsets(extra) {
+        formEls.$FIELDSET_YOUR_DETAILS.addClass(FIELDSET_COLLAPSED);
+        formEls.$FIELDSET_PAYMENT_DETAILS.addClass(FIELDSET_COLLAPSED);
+        formEls.$FIELDSET_REVIEW.addClass(FIELDSET_COLLAPSED);
+        if(extra) {
+            extra.removeClass(FIELDSET_COLLAPSED);
+        }
+    }
+
+    function submitHelper($actionEl, callbackFn) {
+        if($actionEl.length) {
+            bean.on($actionEl[0], 'click', function (evt) {
+                evt.preventDefault();
+                callbackFn();
+            });
+        }
+    }
+
+    function toggleFieldsetsDetails() {
+        submitHelper(formEls.$YOUR_DETAILS_SUBMIT, function(){
+            validations.validatePersonalDetails().then(function (validity) {
+                if(validity.allValid) {
+                    formEls.$FIELDSET_YOUR_DETAILS.addClass(FIELDSET_COLLAPSED).attr(FIELDSET_COMPLETE, '');
+                    formEls.$FIELDSET_PAYMENT_DETAILS.removeClass(FIELDSET_COLLAPSED);
+                    swap(formEls.$EDIT_YOUR_DETAILS, formEls.$EDIT_PAYMENT_DETAILS);
+                }
+            });
+        });
+    }
+
+    function toggleFieldsetsPayment() {
+        submitHelper(formEls.$PAYMENT_DETAILS_SUBMIT, function() {
+
+            var validity = validations.validatePaymentDetails({
```

Comment:
Perhaps we could name these object keys so that they are consistent with the actual form input fields. These way we could just call `reqwest.serialize` on the relevant field set, and pass in the appropriate data to the validation function.